tests/datatypes/test_slice.py

```diff
@@ -75,6 +75,65 @@ def __index__(self):
             print("x[:C(5):C(2] = ", x[:C(5):C(2)])
             print("x[C(2):C(8):C(2)] = ", x[C(2):C(8):C(2)])
             """)
+        self.assertCodeExecution("""
+            class D(object):
+                def __init__(self, value):
+                    self._value = value
+            x = b"0123456789"
+            print("x[D(1)::1] = ", x[D(1)::1])
+        """, exits_early=True)
+        self.assertCodeExecution("""
+            class D(object):
+                def __init__(self, value):
+                    self._value = value
+            x = b"0123456789"
+            print("x[:D(10):1] = ", x[:D(10):1])
+        """, exits_early=True)
+        self.assertCodeExecution("""
+            class D(object):
+                def __init__(self, value):
+                    self._value = value
+            x = b"0123456789"
+            print("x[::D(1)] = ", x[::D(1)])
+        """, exits_early=True)
+
+    def test_slice_bytes(self):
+        self.assertCodeExecution("""
+            x = b"0123456789"
+            print("x[:] = ", x[:])
+            print("x[5:] = ", x[5:])
+            print("x[:5] = ", x[:5])
+            print("x[2:8] = ", x[2:8])
+
+            print("x[::2] = ", x[::2])
+            print("x[5::2] = ", x[5::2])
+            print("x[:5:2] = ", x[:5:2])
+            print("x[2:8:2] = ", x[2:8:2])
+
+            print("x[:] = ", x[:])
+            print("x[-5:] = ", x[-5:])
+            print("x[:-5] = ", x[:-5])
+            print("x[-2:-8] = ", x[-2:-8])
+
+            print("x[::-2] = ", x[::-2])
+            print("x[-5::-2] = ", x[-5::-2])
+            print("x[:-5:-2] = ", x[:-5:-2])
+            print("x[-2:-8:-2] = ", x[-2:-8:-2])
+            """)
+
```

Comment:
Please explain why the testing of negative indexing is done in separate method. I am unable to understand.